src/vma/netlink/netlink_compatibility.cpp

```diff
@@ -112,13 +112,26 @@ int nl_cache_mngr_compatible_add(struct nl_cache_mngr*	mngr, const char* name, c
 	return err;
 }
 
-int nl_object_get_compatible_msgtype(const struct nl_object* obj) {
-	return nl_object_get_msgtype(obj);
+in_addr_t nl_object_get_compatible_gateway(struct rtnl_route* nl_route_obj) {
+	struct rtnl_nexthop *nh;
+	nh = rtnl_route_nexthop_n(nl_route_obj, 0);
+	if (nh) {
+		struct nl_addr * addr;
+		addr = rtnl_route_nh_get_gateway(nh);
+		if (addr) {
+			return *(in_addr_t *) nl_addr_get_binary_addr(addr);
```

Comment:
This type is sure for ipv4 address and since we are checking route entry type before parse it, we assure that it of type in-addr_t.